setup-slave.sh

```diff
@@ -27,38 +27,49 @@ instance_type=$(curl http://169.254.169.254/latest/meta-data/instance-type 2> /d
 
 echo "Setting up slave on `hostname`... of type $instance_type"
 
+# Format & mount using ext4, which has the best performance among ext3, ext4, and xfs based
+# on our shuffle heavy benchmark
+function to_ext4 {
+  device=$1
+  mount_point=$2
+  trim_on=$3
+  EXT4_MOUNT_OPTS="defaults,noatime,nodiratime"
+
+  mkdir -p $mount_point
+  
+  if [[ trim_on == true ]]; then
+    echo $device $mount_point ' ext4 defaults,noatime,nodiratime,discard 0 0' >> /etc/fstab
+  fi
+
+  mkfs.ext4 -E lazy_itable_init=0,lazy_journal_init=0 $device
+  mount -o $EXT4_MOUNT_OPTS $device $mount_point
+}
+
 if [[ $instance_type == r3* || $instance_type == i2* || $instance_type == hi1* ]]; then
   # Format & mount using ext4, which has the best performance among ext3, ext4, and xfs based
   # on our shuffle heavy benchmark
-  EXT4_MOUNT_OPTS="defaults,noatime,nodiratime"
   rm -rf /mnt*
-  mkdir /mnt
-  # To turn TRIM support on, uncomment the following line.
-  #echo '/dev/sdb /mnt  ext4  defaults,noatime,nodiratime,discard 0 0' >> /etc/fstab
-  mkfs.ext4 -E lazy_itable_init=0,lazy_journal_init=0 /dev/sdb
-  mount -o $EXT4_MOUNT_OPTS /dev/sdb /mnt
-
-  if [[ $instance_type == "r3.8xlarge" || $instance_type == "hi1.4xlarge" ]]; then
-    mkdir /mnt2
-    # To turn TRIM support on, uncomment the following line.
-    #echo '/dev/sdc /mnt2  ext4  defaults,noatime,nodiratime,discard 0 0' >> /etc/fstab
-    if [[ $instance_type == "r3.8xlarge" ]]; then
-      mkfs.ext4 -E lazy_itable_init=0,lazy_journal_init=0 /dev/sdc      
-      mount -o $EXT4_MOUNT_OPTS /dev/sdc /mnt2
-    fi
-    # To turn TRIM support on, uncomment the following line.
-    #echo '/dev/sdf /mnt2  ext4  defaults,noatime,nodiratime,discard 0 0' >> /etc/fstab
-    if [[ $instance_type == "hi1.4xlarge" ]]; then
-      mkfs.ext4 -E lazy_itable_init=0,lazy_journal_init=0 /dev/sdf      
-      mount -o $EXT4_MOUNT_OPTS /dev/sdf /mnt2
-    fi    
-  fi
+  device_array=(/dev/sdb /dev/sdc /dev/sdd /dev/sde /dev/sdf /dev/sdg /dev/sdh /dev/sdi)
+
+  total=${#device_array[*]}
+ 
+  # To turn TRIM support on, change the third argument to true
```

Comment:
One interesting thing could be to read in the value for TRIM support through an environment variable -- however afaik there is no clean way to pass in new env variables right now without changes to spark_ec2.py, so this could probably wait.
 
@nchammas it might be interesting to consider how to pass these environment variables elegantly with some of the extension work you are doing